### PR TITLE
enhancement: closes issue #642

### PR DESCRIPTION
**Pull request by dafuga**
Closes issue #642

### CODE REVIEW

File: app/modules/handler/containers/Prompt.js [hunk 1]
```diff
@@ -53,12 +54,18 @@ class PromptContainer extends Component<Props> {
     actions.signURI(tx, blockchain, wallet);
   }
   templateURI = () => {
-    const { blockchains, prompt, wallets } = this.props;
-    const { chainId } = prompt;
+    const { blockchains, prompt: oldprompt, wallets } = this.props;
```
Comment (aaroncox) [inline app/modules/handler/containers/Prompt.js]: Why did you rename prompt?

File: app/modules/handler/containers/Prompt.js [hunk 2]
```diff
@@ -53,12 +54,18 @@ class PromptContainer extends Component<Props> {
...
+    const { chainId } = oldprompt;
     // Set the blockchain for this network
     const blockchain = find(blockchains, { chainId });
     // Find the default wallet for this chain (defaults to first at the moment)
-    const defaultWallet = find(wallets, { chainId });
+
+    const specifiedActor = get(prompt, 'req.1.authorization.0.actor');
+    const specifiedPermission = get(prompt, 'req.1.authorization.0.permission');
+    const defaultWallet =
+      find(wallets, { chainId, account: specifiedActor, authorization: specifiedPermission }) ||
```
Comment (aaroncox) [inline app/modules/handler/containers/Prompt.js]: It'd be cleaner if you just named the variables `account` and `authorization`.

```
    const account = get(prompt, 'req.1.authorization.0.actor');
    const authorization = get(prompt, 'req.1.authorization.0.permission');
    const defaultWallet =
      find(wallets, { chainId, account, authorization }) ||
```

**Comment by aaroncox** [comment]
Other than the two things above - looks like it should work. You tested it?

File: app/modules/handler/containers/Prompt.js (hunk 1 above)
Comment (dafuga) [reply app/modules/handler/containers/Prompt.js]: Renamed the variable while testing and forgot to revert the change.

File: app/modules/handler/containers/Prompt.js (hunk 2 above)
Comment (dafuga) [reply app/modules/handler/containers/Prompt.js]: good point!

**Comment by dafuga** [comment]
Yeah just tested it with the following structure and it worked:
```actions: [{
      account: 'eosio',
      name: 'bidname',
      authorization: [{ actor: 'teamgreymass', permission: 'active' }],
      data: {
        bidder: '............1',
        bid: `${parseFloat(biddingAmount).toFixed(4)} EOS`,
        newname: biddingName
      }
    }]```
I also made the two changes you requested :D

**Comment by aaroncox** [comment]
Awesome! I'll have this merged in for the next Anchor release then :D
